### PR TITLE
Chitter Challenge

### PR DESCRIPTION
**Pull request by kateloschinina**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
   gem 'rake'
   gem 'coveralls', require: false
-end
+  gem 'rubocop-rspec'
+  gem 'rubocop'
+  gem 'factory_girl'
+#  gem 'waiting_rspec_matchers'
+# end
 
+# group :development do
```
Comment (neoeno) [inline Gemfile]: Did you work out why these groups were a problem?

File: app/app.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+require_relative 'data_mapper_setup'
+require 'data_mapper'
+
+class Chitter < Sinatra::Base
+
+  register Sinatra::Flash
+  enable :sessions
+  # use Rack::Session::EncryptedCookie,
+  #   secret: 'ebb09be68821e5f929cce9f98c86efb4daee51287d42d840b388a112d7b3cda8'
```
Comment (neoeno) [inline app/app.rb]: It's hard to tell why this is here and why it's commented out. I'd think about what it does to readability when there's commented out code around.

File: app/app.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  end
+
+  get '/signup' do
+    @user_email = session[:email] ? session[:email] : nil
+    erb :signup
+  end
+
+  post '/tosignup' do
+    redirect to '/signup'
+  end
```
Comment (neoeno) [inline app/app.rb]: It's difficult to work out why this is here.

File: app/app.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+  post '/loginback' do
+    redirect to '/'
+  end
+
+  # start the server if ruby file executed directly
+  run! if app_file == $0
+end
+
+ChitterApp.run! if $0 == 'app/app.rb'
```
Comment (neoeno) [inline app/app.rb]: Which one of these final two lines does something? Or are they both important?

File: app/views/index.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<h2>Would you like to login?</h2>
+<form action = 'login' method = 'post'>
+  <label for 'email'>
+    your email address:
+    <input type = 'email' name = 'email' pattern='[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$' required>
```
Comment (neoeno) [inline app/views/index.erb]: Nice! Good effort on the validation here

File: app/views/index.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  </label>
+  <label for 'password'>
```
Comment (neoeno) [inline app/views/index.erb]: I'd read up on the use of the `for` attribute here.

File: app/views/signup.erb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    your email address:
+    <input type = 'email' name = 'email' value = '<%=@user_email%>' pattern='[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$' required>
+  </label>
+  <label for 'password'>
+    your password:
+    <input type = 'password' name = 'password' required>
+  </label>
+  <label>
+    and please confirm your password:
+    <input type = 'password_confirmation' name = 'password_confirmation' required>
```
Comment (neoeno) [inline app/views/signup.erb]: image

Why might this be?

File: app/public/style.css [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    font-size: 3em;
+    font-weight: 300;
+    outline: none;
+}
+
+label {
+  width: 100%;
+  display: block;
+  text-align: center;
+  margin-top: 1em;
```
Comment (neoeno) [inline app/public/style.css]: Consistent indentation is still important in CSS sadly! Compare lines 102 and 108.

**Comment by kateloschinina** [comment]
Hi Kay!

Thanks a lot for your feedback, I glanced - and so far all makes perfect sense. I will try to go through it in more details and amend the code accordingly later this week (now trying to focus on switching to JS).

I was ill today but hope to recover by tomorrow. Let me know when is good for you to chat about feedback/approaches and behaviour.

**Pull request closed without merging**